### ISSUE TITLE
kqueue is no longer experimental

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5459]
|[<img alt="oberstet's avatar" src="https://avatars.githubusercontent.com/u/233340?s=50" width="50" height="50">](https://github.com/oberstet)| @oberstet reported|
|-|-|
|Trac ID|trac#5459|
|Type|enhancement|
|Created|2012-01-24 00:03:20Z|

[#1918](https://github.com/twisted/twisted/issues/1918) has been resolved, so kqueue reactor is no longer experimental, which should be documented/announced. See [doc/core/howto/choosing-reactor.xhtml](doc/core/howto/choosing-reactor.xhtml).

Attachments:

* [twisted-5459.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/da4/da4bb577ce14531ff7a11e45e038be0991a3b6f8/ab9f38840b05cacaa43fb9a4906793072e4e61f4.patch) (922 bytes) - added by thijs on 2013-02-23 14:55:18Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5459 5459
type__enhancement enhancement
reporter__oberstet oberstet
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1327363400000000 1327363400000000
changetime__1363679386000000 1363679386000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

Likely true, however also note that we do not yet have a FreeBSD builder on the supported list.  When we do, that might be the ideal time to claim KQueue on FreeBSD is "Stable".

**Comment by twisted-trac** [comment]
|[<img alt="oberstet's avatar" src="https://avatars.githubusercontent.com/u/233340?s=50" width="50" height="50">](https://github.com/oberstet)<a name="note_3"></a>|@oberstet commented|
|-|-|

Replying to [exarkun](#note_1):
> Likely true, however also note that we do not yet have a FreeBSD builder on the supported list.  When we do, that might be the ideal time to claim KQueue on FreeBSD is "Stable".

Yep, agreed.

As discussed via email, and to document it here:

I am (for now) committed to maintain the FreeBSD slave

[http://buildbot.twistedmatrix.com/buildslaves/bot-tobias-1]

which has the builder

[http://buildbot.twistedmatrix.com/builders/freebsd-8.2-i386]

And (thanks!) it's now in

[http://buildbot.twistedmatrix.com/supported-builders.txt]

----

Slightly related, I have asked therve whether [#1918](https://github.com/twisted/twisted/issues/1918) could be included in Twisted 12.0 despite the latter already being on release road .. 

''"It's a tough question, but I think I'll say no. There are a couple of tickets that I'd like to see fixed ([#5458](https://github.com/twisted/twisted/issues/5458), [#5459](https://github.com/twisted/twisted/issues/5459)), and it'd be nice to fix the OS X build with it."
''

So it will not land before 12.x x>0 ..

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_6"></a>|@jerub set owner to @thijstriemstra|
|-|-|

Patch accepted without comment. Thank you.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [37667]) Apply twisted-5459.patch: Update the documentation of the kqueue reactor and mention that it is no longer experimental.

Author: thijs
Reviewer: jerub
Fixes: [#5459](https://github.com/twisted/twisted/issues/5459)
